src/Microsoft.Performance.SDK/Processing/IApplicationEnvironmentV2.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using Microsoft.Performance.SDK.Auth;
+
+namespace Microsoft.Performance.SDK.Processing
+{
+    /// <summary>
+    ///     Extends <see cref="IApplicationEnvironment"/> to provide additional functionality.
+    /// </summary>
+    [Obsolete("This interface will be removed in version 2.0 of the SDK. It is OK to use this interface in version 1.x of the SDK.")]
+    public interface IApplicationEnvironmentV2
```

Comment:
>IApplicationEnvironmentV2 [](http://example.com/codeflow?start=21&length=25)

are we using a 'V' for version in interfaces? (instead of 'IApplicationEnvironment2')?

Reply:
I wonder if this shouldn't extend IApplicationEnvironment, but should just be a different interface that is implemented by ApplicationEnvironment.

Reply:
That wouldn't work because plugins only receive `IApplicationEnvironment` instance (not concrete `ApplicationEnvironment` instances) and we need a way for them to call this method. Currently that's with the IAppEnv extension method this PR adds.